neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/embedded/impl/EmbeddedNeo4jGraphDatabaseFactory.java

```diff
@@ -151,20 +153,24 @@ static void shutdownGraphDatabaseService(GraphDatabaseService neo4jDb) {
 				GraphDatabaseFacade graphDatabaseFacade = factoryHolder.getGraphDatabaseFacade();
 				try {
 					GraphDatabaseFacade.SPI spi = null;
-					Object obj = getPrivateField( GraphDatabaseFacade.class.getDeclaredField( "spi" ), graphDatabaseFacade );//
+					Object obj = getPrivateField( GraphDatabaseFacade.class.getDeclaredField( "spi" ), graphDatabaseFacade );
 					spi = (GraphDatabaseFacade.SPI) obj;
 					LOG.debugf( " isInOpenTransaction: %s", spi.isInOpenTransaction() );
 					if ( spi.isInOpenTransaction() ) {
-						LOG.debugf( " currentTransaction: %s", spi.currentTransaction() );
-						//LOG.warnf( "close transaction %s", spi.currentTransaction().getTransactionId() );
-						LOG.warnf( "transaction  started %s", spi.currentTransaction().startTime() );
+						LOG.warnf( "currentTransaction: %s", spi.currentTransaction() );
+						LOG.warnf( "transaction started %s", spi.currentTransaction().startTime() );
+						LOG.warn( "hang transactions  will commited and closed .." );
 						spi.currentTransaction().success();
 						spi.currentTransaction().closeTransaction();
```

Comment:
This code is still closing a transaction when the factory is shut down. You are  coimmitting a transaction under the hood without a user knowing about it. What if the transaction is supposed to be rolled back because something else needs to be done for the user workflow to be correct? Also the transaction could already be in a `failure()` state but you are resetting it with `success()`. I mean, maybe I'm missing something here but this does not seem correct.

Reply:
Okey @DavideD . I will throw Exception .

Reply:
fixed.